packages/contracts/deploy/helpers.ts

```diff
@@ -46,5 +53,80 @@ export async function updateActiveContractsJSON(payload: {
   );
 }
 
+export function ensLabelHash(label: string, ethers: any): string {
+  return ethers.utils.id(label);
+}
+
+export function ensDomainHash(name: string, ethers: any): string {
+  return ethers.utils.namehash(name);
+}
+
+export async function setupENS(hre: HardhatRuntimeEnvironment): Promise<any> {
+  const {deployments, getNamedAccounts, ethers} = hre;
+  const {deploy} = deployments;
+  const {deployer} = await getNamedAccounts();
+
+  const ensRet = await deploy('ENSRegistry', {
+    from: deployer,
+    log: true,
+  });
+  const ensRegistryAddress: string = ensRet.receipt?.contractAddress || '';
+
+  const ensResolverRet = await deploy('PublicResolver', {
+    from: deployer,
+    args: [ensRegistryAddress, ethers.constants.AddressZero],
+    log: true,
+  });
+  const ensResolverAddress: string =
+    ensResolverRet.receipt?.contractAddress || '';
+
+  // setup resolver
+  const ensRegistryContract = await ethers.getContractAt(
+    'ENSRegistry',
+    ensRegistryAddress
+  );
+  const ensResolverContract = await ethers.getContractAt(
+    'PublicResolver',
+    ensResolverAddress
+  );
+  await ensRegistryContract.setSubnodeOwner(
+    ensDomainHash('', ethers),
+    ensLabelHash('resolver', ethers),
+    deployer
+  );
+
+  const resolverNode = ensDomainHash('resolver', ethers);
+
+  await ensRegistryContract.setResolver(resolverNode, ensResolverAddress);
+  await ensResolverContract['setAddr(bytes32,address)'](
+    resolverNode,
+    ensResolverAddress
+  );
+
+  await ensRegistryContract.setSubnodeOwner(
+    ensDomainHash('', ethers),
+    ensLabelHash('eth', ethers),
+    deployer
+  );
+
+  await ensRegistryContract.setSubnodeOwner(
+    ensDomainHash('eth', ethers),
+    ensLabelHash('dao', ethers),
+    deployer
+  );
```

Comment:
This only works for `dao.eth` (mainnet). Is it possible/reasonable to make it parameterizable to account for other chains/environments?